pkg/steps/artifacts.go

```diff
@@ -505,3 +511,65 @@ func allPodContainerNamesWithArtifacts(pod map[string]interface{}) map[string]st
 	}
 	return names
 }
+
+func gatherContainerLogsOutput(podClient PodClient, artifactDir, namespace, podName string) error {
+	var validationErrors []error
+	list, err := podClient.Pods(namespace).List(meta.ListOptions{FieldSelector: fields.Set{"metadata.name": podName}.AsSelector().String()})
+	if err != nil {
+		return fmt.Errorf("could not list pod: %v", err)
+	}
+	pod := &list.Items[0]
+
+	if pod.Annotations["ci-operator.openshift.io/containers-logged-on-failure"] == "" {
+		return nil
+	}
+
+	if err := os.MkdirAll(artifactDir, 0750); err != nil {
+		return fmt.Errorf("unable to create directory %s: %v", artifactDir, err)
+	}
+
+	var containersToLog []string
+	if names := pod.Annotations["ci-operator.openshift.io/containers-logged-on-failure"]; len(names) > 0 {
```

Comment:
I would just make this be per pod: `ci-operator.openshift.io/save-container-logs`

Reply:
So, we don't have to be selective? Just `ci-operator.openshift.io/save-container-logs: "true"` and save the output for all the containers of the pod?

Reply:
yes